### DISCUSSION TITLE
Debug output in UI?

### DISCUSSION THREAD

**Discussion by kentcdodds** [johnlindquist/kit#356] [Errors]
![image](https://user-images.githubusercontent.com/1500684/126346137-d8cd3daf-aefe-4103-b023-7d61f4403e1e.png)

![Untitled](https://user-images.githubusercontent.com/1500684/126346625-87195a60-95e4-45f3-9349-79f35136b3cf.gif)

Any idea why I'm seeing this?

**Comment by johnlindquist** [comment]
It's intentional (and cleaned up a bit in the build beta.31 from a couple minutes ago). Incoming long answer which I'll move to docs 😅

## Why

1. Keep consistent with terminal

I've never liked the `setHint` or `setPlaceholder` solutions to show messages, because they go against what works in the terminal.

2. Users wanted to see what's happening

The first "how do I" question I always get is "how do I see what my script is doing?". Now I can just answer "use `console.log`" 😉

3. It works nicely with `zx`

Kit has bundled `zx` for a while, but now it's way easier to see what it's doing since `zx` logs its commands to the console:

```js
let { stdout } = await $`cd ~/projects && ls`

await arg("Pick a dir", stdout.split("\n"))
```
![CleanShot 2021-07-20 at 10 20 05](https://user-images.githubusercontent.com/36073/126359867-5d40874b-1979-4635-8512-5074aec759bc.png)

4. `console.clear` works

If you want to "hide" the log component:

```js
console.log("Hello!")
let x = await arg("Log should show")
console.clear()
let y = await arg("Log is gone")
console.log(`${x} ${y}`)
await arg("Log is back")
```

5. Log persists between components (unless cleared)

```js
let x = await arg("Type something")
console.log(x)
let y = await textarea({
  placeholder: "Type something else",
})
console.log(y)
let num = await arg("Pick something", [
  "one",
  "two",
  "three",
])
console.log(num)
await wait(1000)
```

6. Works with chalk

```js
console.log(chalk`Request {green passed}!`)
console.log(chalk`Request {red failed}...`)

await arg()
```
![CleanShot 2021-07-20 at 10 28 45](https://user-images.githubusercontent.com/36073/126361171-43b669ee-fd8b-4334-98c0-23b5346cc283.png)

**Reply by kentcdodds** [reply]
Neat! I didn't realize that I had `console.time` still in my script so it look like a bug or something 😅

**Comment by andrefagl** [comment]
Hello guys,

I'm starting to play around with kit and for my first script, I thought it would be cool to modify the `Open Project` script to display the current git branch or directory path in the list description.

To accomplish this, I'm running the following `zx` command `$'git --git-dir ${currentDir}/.git branch --show-current/'` and as expected the log is showing the `zx` stdout. 

Do you know if there is a way to hide the log component while my script runs?

Thank you for your amazing work @johnlindquist  ❤️ 

<img width="768" alt="CleanShot 2022-03-17 at 22 19 44@2x" src="https://user-images.githubusercontent.com/479683/158905379-75ba4804-c008-42d7-a6cb-da675f864745.png">

**Reply by johnlindquist** [reply]
@andrefagl Apologies for the delayed response, been moving into a new house and setting up a new office. 😅

You can disable logging in the UI with:

```js
// Log: false
```

Or if you want to log something, then hide it:
```js
console.clear() // closes the log UI until the next console.log()
```

**Reply by andrefagl** [reply]
Hi @johnlindquist, thanks for taking the time to help with this! 

`// Log:  false` fits perfectly well for my use case ;)

GL with the moving 💪
